platform/entanglement/src/services/CopyTask.js

```diff
@@ -137,23 +138,37 @@ define(
          */
         CopyTask.prototype.copy = function(originalObject, originalParent) {
             var self = this,
-                modelClone = {
-                id: uuid(),
-                model: cloneObjectModel(originalObject.getModel()),
-                persistenceSpace: originalParent.hasCapability('persistence') && originalParent.getCapability('persistence').getSpace()
-            };
-
-            return this.$q.when(originalObject.useCapability('composition')).then(function(composees){
-                self.deferred.notify({phase: "preparing"});
-                //Duplicate the object's children, and their children, and
-                // so on down to the leaf nodes of the tree.
-                return self.copyComposees(composees, modelClone, originalObject).then(function (){
-                    //Add the clone to the list of clones that will
-                    //be returned by this function
-                    self.clones.push(modelClone);
-                    return modelClone;
+                clone;
+
+            //Check if the type of the object being copied allows for
+            // creation of new instances. If it does not, then a link to the
+            // original will be created instead.
+            if (this.policyService.allow("creation", originalObject.getCapability("type"))){
+                //create a new clone of the original object. Use the
+                // creation capability of the targetParent to create the
+                // new clone. This will ensure that the correct persistence
+                // space is used.
+                clone = this.parent.hasCapability("instantiation") && this.parent.useCapability("instantiation", cloneObjectModel(originalObject.getModel()));
```

Comment:
It would be safe to omit the `hasCapability` check here - we're going to fail anyway if this doesn't exist so may as well fail fast. `instantiation` is in core so it's safe to assume it will be there, and if that assumption is ever invalidated I think the check would belong in `CopyService#validate`